src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const copy = {};
+
+  Object.assign(copy, state);
+
+  for (const i of actions) {
```

Comment:
The variable 'i' is not descriptive. Consider using a more descriptive variable name such as 'action'.